File: leetcode/0724-find-pivot-index/pivotIndex.ts
function pivotIndex(nums: number[]): number {
  let index = -1;
  let sum = 0;
  let leftSum = 0;

  for (const num of nums) {
    sum = sum + num;
  }

  for (const [i, num] of nums.entries()) {
    const rightSum = sum - leftSum - num;

    if (rightSum === leftSum) {
      index = i;
      break;
    }

    leftSum = leftSum + num;
  }

  return index;
}

export default pivotIndex;

File: leetcode/0724-find-pivot-index/pivotIndex.go
package leetcode

func pivotIndex(nums []int) int {
	var index = -1
	var sum = 0
	var leftSum = 0
	for _, num := range nums {
		sum = sum + num
	}

	for i, num := range nums {
		var rightSum = sum - leftSum - num

		if rightSum == leftSum {
			index = i
			break
		}

		leftSum = leftSum + num
	}

	return index
}
